src/libraries/System.Data.Common/src/System/Xml/XmlDataDocument.cs

```diff
@@ -2553,8 +2551,21 @@ public XmlDataDocument(DataSet dataset) : base(new XmlDataImplementation())
 
         internal XmlDataDocument(XmlImplementation imp) : base(imp)
         {
-        }
-
+            // This constructor is used by XmlDataImplementation.CreateDocument(), which
+            // exposes it as XmlDocument. The methods using these fields are never called.
+            _dataSet = null!;
```

Comment:
Do we need both the `= null!` here and in the field initializers at the top of the class?

```C#
 private DataSet _dataSet = null!;
```

Reply:
My apologies, I thought I reverted them already, which I didn't. Fixed.